doc/tutorials/using_zephyr_as_uos.rst

```diff
@@ -0,0 +1,88 @@
+.. _using_zephyr_as_uos:
+
+Using Zephyr as User OS
+#######################
+
+The tutorial describes the steps to run zephyr as User OS on ACRN hypervisor. We are using
+Kaby Lake-based NUC model NUC7i5DNH and you can try other platform ACRN supported as well.
+
+Introduction to Zephyr
+**********************
+
+The Zephyr is a scalable real-time operating-system (RTOS) supporting multiple hardware architecture,
+optimized for resource constrained devices, and built with safety and security in mind.
+
+Steps for Using Zephyr as User OS
+*********************************
+
+#. Build Zephyr and grub2 boot loader image
+
+   First of all, please follow the `Development Environment Setup on Linux
+   <https://docs.zephyrproject.org/latest/getting_started/installation_linux.html>`_ to
+   setup Zephyr development environment. Then follow the `Board-Specfic Support
+   <https://docs.zephyrproject.org/latest/boards/index.html>`_ to build the acrn supported zephyr image ``zephyr.elf``.
+
+   Build grub2 boot loader image
+
+   .. code-block:: none
+
+      $ git clone https://github.com/zephyrproject-rtos/zephyr.git
+      $ cd $ZEPHYR_BASE
+      $ ./boards/x86/common/scripts/build_grub.sh x86_64
```

Comment:
I'm getting an error on my system (`Clear Linux version: 27670`) when running this step:
```
In file included from grub-core/partmap/gpt.c:26:
./include/grub/gpt_partition.h:79:1: error: alignment 1 of ‘struct grub_gpt_partentry’ is less than 8 [-Werror=packed-not-aligned]
 } GRUB_PACKED;
 ^
cc1: all warnings being treated as errors
make[2]: *** [Makefile:6490: grub-core/partmap/libgrubkern_a-gpt.o] Error 1
make[2]: *** Waiting for unfinished jobs....
```

I have tried with both Zephyr SDK 0.9.5 and 0.10.0, will try shortly upgrading my Clear Linux host to see if it makes any difference.

Have you seen this error before?

Reply:
I met the same error on clearlinux. But it works on ubuntu-16.04.
How about add one note here to remind user to use ubuntu-1604?

Reply:
Good to know it wasn't just me :-)

I have filed a bug in Zephyr to track the issue on Clear Linux: https://github.com/zephyrproject-rtos/zephyr/issues/14289

Because of that issue, I used Fedora 29 to build it but had problems with it too. It was easy enough to resolve them though, a PR is pending with the fixes here: https://github.com/zephyrproject-rtos/zephyr/pull/14281

I have not tested other versions but I would simply add a note for now indicating there is a known issue with Clear Linux (and a link to it).

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/14468 landed recently in Zephyr. Fedora 29 and Clear Linux can now build the GRUB2 firmware image correctly, so no need to insert any note anymore :-)